### DISCUSSION TITLE
Troubleshooting ECONNREFUSED Error When Connecting Flowise and LocalAI

### DISCUSSION THREAD

**Discussion by nattyraz** [mudler/LocalAI#576] [Q&A]
Hello,

I've been working on setting up Flowise and LocalAI locally on my machine using Docker. Although I'm not an expert in coding, I've managed to get some systems running locally. I've ensured the appropriate model files are in the /models directory and the setup for Pinecone and OpenAI API is correct.

Both Flowise and LocalAI are accessible at their respective addresses: http://localhost:3000/ for Flowise and http://localhost:8080/ for LocalAI. However, I'm encountering an issue when trying to connect to the LocalAI server from the ChatLocalAI module in the Conversational Retrieval QA Chain on Flowise. Specifically, I'm receiving an ECONNREFUSED error on 127.0.0.1:8080 when I send a request.

To troubleshoot, I've tested this setup on two different PCs, but I'm consistently encountering the same error. Interestingly, I was able to communicate with a model directly in VSCode, but I've been unsuccessful in doing so within Flowise.

Any guidance on how to resolve this issue would be greatly appreciated.

**Comment by mudler** [comment]
If you are running the docker-compose example, you should specify http://api:8080 as address. See the image over here:
![](https://user-images.githubusercontent.com/2420543/242028891-02458782-0549-4131-971c-95ee56ec1af8.png)

**Reply by nattyraz** [reply]
hello, i not better ... i get this message now "getaddrinfo EAI_AGAIN api" ??

**Comment by srsettur** [comment]
Hi, I am facing similar issue but with http://127.0.0.1:8080/v1/chat/completions. 
http://127.0.0.1:8080/v1/models/ worked well and is giving loaded model name. 

Model: ggml-gpt4all-j.bin
Temperature: 0.7
Input
Get me a artistic peom on rain
Error
error
{
  "cause": {
    "code": "ECONNREFUSED",
    "type": "system",
    "errno": "ECONNREFUSED",
    "message": "request to http://127.0.0.1:8080/v1/chat/completions failed, reason: connect ECONNREFUSED 127.0.0.1:8080"
  },
  "retriesLeft": 0,
  "attemptNumber": 7
}

can someone help me on this?
